### ISSUE TITLE
KOReader doesn't use https even when it's possible to do so

### ISSUE THREAD

**Issue by 0rzech** [koreader/koreader#9685]
* KOReader version: 2022.10 "Muhara"

#### Issue

KOReader ota url and some dictionary urls are hardcoded to http, eg:

* `ota_servers` contains `http://ota.koreader.rocks/` instead of `https://ota.koreader.rocks/`
* `dictionaries` contains `http://build.koreader.rocks/download/dict/*` instead of `https://build.koreader.rocks/download/dict/*`

IMHO, these should be changed to https. And if ota must be available unencrypted, then we should add https ota as another option in the list.

**Comment by poire-z** [comment]
(What if we want to advertize KOReader to the man in the middle ? I assure you he will stop using tcpdump once he has tried KOReader ! :) )

**Comment by Frenzie** [comment]
zsync doesn't work with https and the overhead is not insignificant, but since we switched to zsync2 fairly recently and optimized our requests it might work well enough. Have you tried it?

**Comment by 0rzech** [comment]
No, I haven't. Is it enough to to edit lua file with ota urls and just try to update?

**Comment by Frenzie** [comment]
Pretty much. Don't forget to restart the program after editing (or better yet, just close it while you're editing the files). ^_^

**Comment by poire-z** [comment]
Is the impact on battery of transfering 30Mb over SSL vs. plain on our little devices negligible?

**Comment by 0rzech** [comment]
I did this:

```diff
***************
*** 23,28 ****
--- 23,29 ----
      -- NOTE: Each URL *MUST* end with a /
      ota_servers = {
          "http://ota.koreader.rocks/",
+         "https://ota.koreader.rocks/",
          --[[
          -- NOTE: Seems down? Ping @chrox ;).
          "http://vislab.bjmu.edu.cn/apps/koreader/ota/",
```

and the check was way too fast to use a stopwatch: <1-2s I think. 😉

If you're worried about possible negative performance impact, then I suggest just adding https as non-default option for people preferring to use it anyway. 😉

**Comment by NiLuJe** [comment]
My concern is not with the check, because that's left to LuaSocket, but with the actual zsync process.

I don't quite recall if the build is setup to handle certs properly (... or if it even works properly.). As @Frenzie said, it at least has *a* chance to work since we switched to zsync2, but whether it does so reliably is... another kettle of fish ;).

**Comment by NiLuJe** [comment]
> I don't quite recall if the build is setup to handle certs properly

Just checked, it is. yay, past me! :D.

**Comment by 0rzech** [comment]
I thought the update check starts zsync anyway. 😉 Well, I understand the uncertainty, unfortunately. 😜

I can keep the setting and tell you how it works. But it will be overridden on each update, which kinda sucks. 😉

And what about dictionaries? Most are https anyways.

**Comment by Frenzie** [comment]
> optimized our requests

For Cloudflare which behaves similarly to HTTPS wrt small requests I should add.
